File: NoteBookTask/NoteBookTask/CreateNoteBook.cs
namespace NoteBookTask
{
    /// <summary>
    /// Creates notebook
    /// </summary>
    public class CreateNoteBook : ICommand
    {
        NoteBookProvider receiver;

        /// <summary>
        /// Constuctor of class
        /// </summary>
        /// <param name="receiver">
        /// used by command pattern
        /// </param>
        public CreateNoteBook(NoteBookProvider receiver)
        {
            this.receiver = receiver;
        }

        /// <summary>
        /// Create notebook
        /// </summary>
        public void Execute()
        {
            receiver.CreateNoteBook();
        }
    }
}

File: TaskNoteBook/src/main/java/CreateNoteBook.java
/**
 * Creates new NoteBook.
 */
public class CreateNoteBook implements ICommand {
  private NoteBookProvider receiver;

  /**
   * Constructor for this class.
   *
   * @param receiver used by command pattern
   */
  public CreateNoteBook(NoteBookProvider receiver) {
    this.receiver = receiver;
  }

  /**
   * Creates new NoteBook.
   */
  @Override
  public void execute() {
    receiver.createNoteBook();
  }
}
